substrate/frame/contracts/src/xcm/adapter.rs

```diff
@@ -0,0 +1,124 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//! Adapter exposes an implementation of the [`XCM`] trait using pallet-xcm.
+
+use crate::{
+	xcm::{CallOf, WeightInfo, XCM},
+	AccountIdOf, Box, Config, RawOrigin,
+};
+use frame_support::{pallet_prelude::DispatchResultWithPostInfo, weights::Weight};
+use frame_system::pallet_prelude::BlockNumberFor;
+use pallet_xcm::WeightInfo as XcmWeightInfo;
+use sp_runtime::{DispatchError, DispatchResult};
+use xcm::{v3::MultiLocation, VersionedMultiLocation, VersionedXcm};
+use xcm_executor::traits::{QueryHandler, QueryResponseStatus};
+
+/// An implementation of the [`XCM`] trait using pallet-xcm.
+pub struct XCMAdapter<T: pallet_xcm::Config>(sp_std::marker::PhantomData<T>);
```

Comment:
Nit: standard Rust syntax is to capitalize only the first letter in acronyms, so:

```suggestion
pub struct XcmAdapter<T: pallet_xcm::Config>(sp_std::marker::PhantomData<T>);
```